Source/DafnyServer/CounterExampleGeneration/DafnyModelState.cs

```diff
@@ -142,18 +145,16 @@ public class DafnyModelState {
         if (val == null) {
           continue; // This variable has no value in the model, so ignore it.
         }
+
         var vn = DafnyModelVariableFactory.Get(this, val, v, duplicate: true);
-        if (curVars.ContainsKey(v)) {
-          Model.RegisterLocalValue(vn.Name, val);
-        }
```

Comment:
I don't know what this was doing before, but can you please explain me why you could delete it safely?

Reply:
This remained from the original version that existed before I refactored the counterexample extraction code. I think its purpose must have been some form of field name preprocessing but at this point it is dead code. The only thing that [`DafnyModel.RegisterLocalValue`](https://github.com/dafny-lang/dafny/blob/dc6e7f086a2fe4860d8885ad7a0ab80ac13a2b8e/Source/DafnyServer/CounterExampleGeneration/DafnyModel.cs#L824-L830) does is update the `localValue` dictionary, which is never used elsewhere, so I removed it. The two methods that `RegisterLocalValue` calls, which are `CompareFieldNames` and `GetNumber`, are static, non side-effecting and operate on strings, so I also remove these as part of this PR.

Reply:
Fair enough, thanks for the explanation.